src/pages/TokenPage/TokenPage.tsx

```diff
@@ -0,0 +1,22 @@
+import React, { useEffect } from "react";
+import { useHistory, useLocation } from "react-router-dom";
+import { useStore } from "../../Store";
+import { PATHS } from "../../Routes";
+
+const TokenPage: React.FC = () => {
+  const { setApiToken } = useStore();
+  const history = useHistory();
+  const location = useLocation();
+
+  useEffect(() => {
+    const params = new URLSearchParams(location.search);
+    if (params.has("token")) {
+      setApiToken(params.get("token"));
+      history.replace(PATHS.HOME_PAGE);
```

Comment:
Note to other reviewers: Here are the docs for `history.replace`: https://v5.reactrouter.com/web/api/history